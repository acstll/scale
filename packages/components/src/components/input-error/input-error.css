/**
 * @prop --input-error-font-size: Font size of the input-error
 * @prop --input-error-font-family: Font family of the input-error
 * @prop --input-error-padding-top: Padding top of the input-error
 * @prop --input-error-color: Color of the input-error
 */

.input-error {
  font-size: var(--input-error-font-size, 0.8rem);
  font-family: var(--input-error-font-family, unset);
  padding-top: var(--input-error-padding-top, 0.5rem);
  color: var(--input-error-color, lightcoral);
}
