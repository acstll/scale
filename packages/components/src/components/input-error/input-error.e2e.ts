import { newE2EPage } from '@stencil/core/testing';

describe('t-input-error', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-input-error/>');
    const element = await page.find('t-input-error');
    expect(element).toHaveClass('hydrated');
  });
});
