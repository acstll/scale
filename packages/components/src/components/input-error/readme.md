# t-input-text



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                 | Type     | Default |
| ------------- | -------------- | --------------------------- | -------- | ------- |
| `customClass` | `custom-class` | (optional) Input text class | `string` | `''`    |
| `theme`       | `theme`        | (optional) Input text theme | `string` | `''`    |


## CSS Custom Properties

| Name                        | Description                    |
| --------------------------- | ------------------------------ |
| `--input-error-color`       | Color of the input-error       |
| `--input-error-font-family` | Font family of the input-error |
| `--input-error-font-size`   | Font size of the input-error   |
| `--input-error-padding-top` | Padding top of the input-error |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
