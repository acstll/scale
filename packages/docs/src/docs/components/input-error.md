---
title: Input error
description: Input error element
url: /docs/input-error
contributors:
  - eeegor
---

# Input error

Input error component

selector: `t-input-error`

## Default

```javascript
<t-input-error>Required</t-input-error>
```

<div class="demo-container">
  <t-input-error>Required</t-input-error>
</div>
