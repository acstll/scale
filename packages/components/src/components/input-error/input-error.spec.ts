import { newSpecPage } from '@stencil/core/testing';
import { InputError } from './input-error';

describe('Input Error', () => {
  let element;
  beforeEach(async () => {
    element = new InputError();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [InputError],
      html: `<t-input-error>Error</t-input-error>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle theme css class', () => {
    element.theme = 'default';
    expect(element.getCssClassMap()).toContain('input-error--theme-default');
  });

  it('should have a default css class', () => {
    expect(element.getCssClassMap()).toContain('input-error');
  });
});
