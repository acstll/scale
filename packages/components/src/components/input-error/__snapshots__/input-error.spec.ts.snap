// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Error should match snapshot 1`] = `
<t-input-error>
  <mock:shadow-root>
    <div class="input-error">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Error
</t-input-error>
`;
